packages/shared/src/validation/addressValidation.ts

```diff
@@ -28,32 +32,13 @@ const assertArraylikeEqual = <T>(a: ArrayLike<T>, b: ArrayLike<T>) => {
   }
 };
 
-// if we go back to ethers 6 when typechain updates
-// const hexToUint8Array = (hex: string) => {
-//   const withoutPrefix = hex.replace(/^0x/, '');
-//   const padded =
-//     withoutPrefix.length % 2 === 0 ? withoutPrefix : `0${withoutPrefix}`;
-//   const matchArray = padded.match(/.{2}/g);
-//   assert(matchArray, 'matchArray must not be null');
-//   return new Uint8Array(matchArray.map((n) => Number.parseInt(n, 16)));
-// };
-
-// const decodeBase58 = (address: string) => {
-//   const bigint = ethers.decodeBase58(address);
-//   // this regex will always match
-//   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-//   const leadingZeroes = address.match(/^1*/)![0].length;
-//   const hex = `${'0'.repeat(leadingZeroes)}${bigint.toString(16)}`;
-//   return hexToUint8Array(hex);
-// };
-
 const validateP2PKHOrP2SHAddress = (
   address: string,
   network: BitcoinNetwork,
 ) => {
   try {
     // The address must be a valid base58 encoded string.
-    const decoded = ethers.utils.base58.decode(address);
+    const decoded = base58Decode(address);
```

Comment:
it seems like we can use the `base58Decode` method of the `@polkadot/util-crypto` package as replacement

Reply:
Good find, the important thing is that leading `1`s are properly parsed to a leading zero byte:

```
> base58Decode('123456')
Uint8Array(4) [ 0, 178, 201, 185 ]
```